src/app/contributors/store/contributors.store.ts

```diff
@@ -0,0 +1,55 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { inject, Injectable } from '@angular/core';
+import {
+  ComponentStore,
+  OnStateInit,
+  tapResponse,
+} from '@ngrx/component-store';
+import { pipe, switchMap, tap } from 'rxjs';
+import { GithubService } from 'src/app/services/github.service';
+import { ContributorsModel } from '../contributors.model';
+
+interface ContributorsStateModel {
+  contributors: ContributorsModel[];
+  isLoading: boolean;
+  error?: Error | null;
+}
+
+@Injectable()
+export class ContributorsStore
+  extends ComponentStore<ContributorsStateModel>
+  implements OnStateInit
+{
+  private githubService = inject(GithubService);
+  constructor() {
+    super({
+      contributors: [],
+      isLoading: false,
+    });
+  }
+
+  ngrxOnStateInit() {
+    this.getContributors();
+  }
+
+  readonly vm = this.selectSignal(({ contributors, isLoading }) => ({
```

Comment:
```suggestion
  readonly contributorsData = this.selectSignal(({ contributors, isLoading }) => ({
```

I'm not confident with this `vm`, `contributorsData` or how do you want

Reply:
in my opinion, some properties like `isLoading` are unrelated to contributors' data. I think `contributorsData ` or `contributors`  are good for selecting contributors from our store.
```typescript
readonly contributors = computed(() => this.state().contributors)
```
 also, in the ngrx documentation used `vm`. so I think it's better that we use the same convention.